### ISSUE TITLE
cloudbeat 8.6.* findings are not being indexed on package version 1.2.*

### ISSUE THREAD

**Issue by kfirpeled** [elastic/cloudbeat#768]
**Describe the bug**
Cloudbeat's findings are not indexed due to `failed to parse field [cloudbeat.commit_time] of type [date]`

**Preconditions**

ELK version: >= 8.7
Agent version: 8.6.*

**To Reproduce**

1. Install latest 1.2.* package `kspm` `self managed`
2. Deploy an agent (I used standalone mode on local kind cluster)

**Expected behavior**

Findings are successfully indexed

**Comment by kfirpeled** [comment]
**UPDATE**
It appears that the format of `commit_time` field has been changed between cloudbeat 8.6 and 8.7 (see  https://github.com/elastic/cloudbeat/pull/641)

In order to support the old formatting I've changed the format of `commit_time` to `yyyy-MM-dd HH:mm:ss Z z||strict_date_optional_time||epoch_millis` while `strict_date_optional_time||epoch_millis` is the default formatting

Testing this locally, I've managed to see the format works properly. However, when trying to upgrade the package fleet doesn't update the mappings :\ I'm not sure why it fails to detect these changes.

When I manually updated the mappings, the transform failed the next time with the same error. That is because the mappings of the target index are not updated. This brings another level of complexity, although our package is set for an auto upgrade, it doesn't kick in after the server's restart.
